README.md
HTTP HANDLER
============

Drupal's module which provide handler in order to support multiple 
Exceptions for 4xx or 5xx HTTP status code.


INTRODUCTION
------------

The HTTP Handler module offer possibility to support many of the
http status return code in Drupal.

For example, when anonymous user try to get content of a "secure" node
it will throw a 401 instead of a 403.
You can customize error page with a content node type like
404 or 403 error page.

 * For a full description of the module, visit the project page:
   https://drupal.org/project/http_handler

 * To submit bug reports and feature suggestions, or to track changes:
   https://drupal.org/project/issues/http_handler


REQUIREMENTS
------------

This module currently does not requires any other modules, but
you should use a Drupal >= 7.10


RECOMMENDED MODULES
-------------------

 * Markdown filter (https://www.drupal.org/project/markdown):
   When enabled, display of the project's README.md help will be rendered
   with markdown.
 * X Autoload (https://www.drupal.org/project/xautoload):
   Provides Drupal-8-style PSR-4.


INSTALLATION
------------

 * Install as you would normally install a contributed Drupal module. Visit:
   https://drupal.org/documentation/install/modules-themes/modules-7
   for further information.


CONFIGURATION
-------------

Customize error's look content in Administration » Configuration » Système.


TROUBLESHOOTING
---------------

 * If error page does not show the correct content.

  * Because by default it show 403 errors, make sure you've clean 
cache before testing.
Headers should be good instead.

FAQ
---

Q: Is the module works with fast_404?

A: Regards to fast_404 it should. But you'll not have all proposed
errors from http_handler module.

MAINTAINERS
-----------

Current maintainers:
 * Adrien Loyant (donaldinou) - https://www.drupal.org/u/donaldinou


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

Types of changes :
- `Added` for new features.
- `Changed` for changes in existing functionality.
- `Deprecated` for soon-to-be removed features.
- `Removed` for now removed features.
- `Fixed` for any bug fixes.
- `Security` in case of vulnerabilities.

## [Unreleased]

## [0.6] - 2019-08-21
### Changed
- CHANGELOG standard.
- phpcs:disable / enable

## [0.5] - 2019-02-13
### Fixed
- PHP7 compatibility

## [0.4] - 2017-08-25
### Changed
- Coding Standard : Remove site_* var in flavor of http_handler_site_*
- Coding Standard : Remove drupal_* functions in flavor of http_handler_drupal_*

### Removed
- LICENSE (drupal.org)

## [0.3] - 2017-08-24
### Added
- 410 Gone error

### Changed
- Coding Standard
- Readme Drupal Template

### Fixed
- hook_help... (Damned to fast implementation)

## [0.2] - 2017-08-18
### Added
- HttpStatusCodeEnum enumerator
- hook_help
- Translations

### Changed
- Change licensing

### Fixed
- PSR2 standard code

## [0.1] - 2017-08-01
### Added
- 400/401/405 http request handler

[Unreleased]: https://git.drupal.org/project/http_handler/compare/7.x-0.6...HEAD
[0.6]: https://git.drupal.org/project/http_handler/compare/7.x-0.5...7.x-0.6
[0.5]: https://git.drupal.org/project/http_handler/compare/7.x-0.4...7.x-0.5
[0.4]: https://git.drupal.org/project/http_handler/compare/7.x-0.3...7.x-0.4
[0.3]: https://git.drupal.org/project/http_handler/compare/7.x-0.2...7.x-0.3
[0.2]: https://git.drupal.org/project/http_handler/compare/7.x-0.1...7.x-0.2
[0.1]: https://git.drupal.org/project/http_handler/tree/7.x-0.1


src/Component/Enum/HttpStatusCodeEnum.php
<?php

namespace Drupal\http_handler\Component\Enum;

/**
 * Enumerator for HTTP Response Status Code.
 *
 *  - HTTP_PROCESSING :                                                RFC2518
 *  - HTTP_MULTI_STATUS :                                              RFC4918
 *  - HTTP_ALREADY_REPORTED :                                          RFC5842
 *  - HTTP_IM_USED :                                                   RFC3229
 *  - HTTP_PERMANENTLY_REDIRECT :                                      RFC7238
 *  - HTTP_I_AM_A_TEAPOT :                                             RFC2324
 *  - HTTP_MISDIRECTED_REQUEST :                                       RFC7540
 *  - HTTP_UNPROCESSABLE_ENTITY :                                      RFC4918
 *  - HTTP_LOCKED :                                                    RFC4918
 *  - HTTP_FAILED_DEPENDENCY :                                         RFC4918
 *  - HTTP_RESERVED_FOR_WEBDAV_ADVANCED_COLLECTIONS_EXPIRED_PROPOSAL : RFC2817
 *  - HTTP_UPGRADE_REQUIRED :                                          RFC2817
 *  - HTTP_PRECONDITION_REQUIRED :                                     RFC6585
 *  - HTTP_TOO_MANY_REQUESTS :                                         RFC6585
 *  - HTTP_REQUEST_HEADER_FIELDS_TOO_LARGE :                           RFC6585
 *  - HTTP_VARIANT_ALSO_NEGOTIATES_EXPERIMENTAL :                      RFC2295
 *  - HTTP_INSUFFICIENT_STORAGE :                                      RFC4918
 *  - HTTP_LOOP_DETECTED :                                             RFC5842
 *  - HTTP_NOT_EXTENDED :                                              RFC2774
 *  - HTTP_NETWORK_AUTHENTICATION_REQUIRED :                           RFC6585
 *
 * @see http_response_code().
 */
class HttpStatusCodeEnum {

  // In lowercase because of SplEnum.
  const __default = self::HTTP_OK; // @codingStandardsIgnoreLine

  const HTTP_CONTINUE = 100;
  const HTTP_SWITCHING_PROTOCOLS = 101;
  const HTTP_PROCESSING = 102;

  const HTTP_OK = 200;
  const HTTP_CREATED = 201;
  const HTTP_ACCEPTED = 202;
  const HTTP_NON_AUTHORITATIVE_INFORMATION = 203;
  const HTTP_NO_CONTENT = 204;
  const HTTP_RESET_CONTENT = 205;
  const HTTP_PARTIAL_CONTENT = 206;
  const HTTP_MULTI_STATUS = 207;
  const HTTP_ALREADY_REPORTED = 208;
  const HTTP_IM_USED = 226;

  const HTTP_MULTIPLE_CHOICES = 300;
  const HTTP_MOVED_PERMANENTLY = 301;
  const HTTP_FOUND = 302;
  const HTTP_SEE_OTHER = 303;
  const HTTP_NOT_MODIFIED = 304;
  const HTTP_USE_PROXY = 305;
  const HTTP_RESERVED = 306;
  const HTTP_TEMPORARY_REDIRECT = 307;
  const HTTP_PERMANENTLY_REDIRECT = 308;

  const HTTP_BAD_REQUEST = 400;
  const HTTP_UNAUTHORIZED = 401;
  const HTTP_PAYMENT_REQUIRED = 402;
  const HTTP_FORBIDDEN = 403;
  const HTTP_NOT_FOUND = 404;
  const HTTP_METHOD_NOT_ALLOWED = 405;
  const HTTP_NOT_ACCEPTABLE = 406;
  const HTTP_PROXY_AUTHENTICATION_REQUIRED = 407;
  const HTTP_REQUEST_TIMEOUT = 408;
  const HTTP_CONFLICT = 409;
  const HTTP_GONE = 410;
  const HTTP_LENGTH_REQUIRED = 411;
  const HTTP_PRECONDITION_FAILED = 412;
  const HTTP_REQUEST_ENTITY_TOO_LARGE = 413;
  const HTTP_REQUEST_URI_TOO_LONG = 414;
  const HTTP_UNSUPPORTED_MEDIA_TYPE = 415;
  const HTTP_REQUESTED_RANGE_NOT_SATISFIABLE = 416;
  const HTTP_EXPECTATION_FAILED = 417;
  const HTTP_I_AM_A_TEAPOT = 418;
  const HTTP_MISDIRECTED_REQUEST = 421;
  const HTTP_UNPROCESSABLE_ENTITY = 422;
  const HTTP_LOCKED = 423;
  const HTTP_FAILED_DEPENDENCY = 424;
  const HTTP_RESERVED_FOR_WEBDAV_ADVANCED_COLLECTIONS_EXPIRED_PROPOSAL = 425;
  const HTTP_UPGRADE_REQUIRED = 426;
  const HTTP_PRECONDITION_REQUIRED = 428;
  const HTTP_TOO_MANY_REQUESTS = 429;
  const HTTP_REQUEST_HEADER_FIELDS_TOO_LARGE = 431;
  const HTTP_UNAVAILABLE_FOR_LEGAL_REASONS = 451;

  const HTTP_INTERNAL_SERVER_ERROR = 500;
  const HTTP_NOT_IMPLEMENTED = 501;
  const HTTP_BAD_GATEWAY = 502;
  const HTTP_SERVICE_UNAVAILABLE = 503;
  const HTTP_GATEWAY_TIMEOUT = 504;
  const HTTP_VERSION_NOT_SUPPORTED = 505;
  const HTTP_VARIANT_ALSO_NEGOTIATES_EXPERIMENTAL = 506;
  const HTTP_INSUFFICIENT_STORAGE = 507;
  const HTTP_LOOP_DETECTED = 508;
  const HTTP_NOT_EXTENDED = 510;
  const HTTP_NETWORK_AUTHENTICATION_REQUIRED = 511;

}
